builtin/providers/google/resource_google_service_account_test.go

```diff
@@ -0,0 +1,139 @@
+package google
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+var (
+	displayName  = "Terraform Test"
+	displayName2 = "Terraform Test Update"
+)
+
+// Test that a service account resource can be created, updated, and destroyed
+func TestAccGoogleServiceAccount_basic(t *testing.T) {
+	accountId := "a" + acctest.RandString(10)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			// The first step creates a basic service account
+			resource.TestStep{
+				Config: fmt.Sprintf(testAccGoogleServiceAccount_basic, accountId, displayName),
```

Comment:
The pattern I've seen is that these get implemented as functions, with `accountId` and `displayName` passed in as arguments and the Config string returned, instead of as `fmt.Sprintf` strings. This seems like a better option to me, as it documents the requirements and should they change after this is reused, it won't build unless you update them everywhere. With the `Sprintf` string, it'll just have weird output.